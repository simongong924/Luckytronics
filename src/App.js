import React, { Component } from 'react';
import { BrowserRouter as Router, Route, Link } from "react-router-dom";
import "bootstrap/dist/css/bootstrap.min.css";
import SignUpForm from './pages/SignUpForm';
import SignInForm from './pages/SignInForm';
import Ticktform from './pages/Ticktform';
import TrackTicket from './pages/trackticket';
import homepageUser from './pages/homepageUser';
//import Navigation from './Navigation';
import Table1 from'./pages/table1';
import './App.css';
import logo from "./logo.png";

  var data = [
  {id: 1, name: 'Gob', value: '2'},
  {id: 2, name: 'Buster', value: '5'},
  {id: 3, name: 'George Michael', value: '4'}
];

class App extends Component {
  render() {
    return (
      <div id="parent">
      <Router>
        <div className="container">

          <nav className="navbar navbar-expand-lg navbar-light bg-light">
            <a className="navbar-brand" href="https://codingthesmartway.com" target="_blank">
              <img src={logo} width="30" height="30" alt="CodingTheSmartWay.com" />
            </a>
            <Link to="/" className="navbar-brand">Ticket System</Link>
            <div className="collpase nav-collapse">
              <ul className="navbar-nav mr-auto">
                <li className="navbar-item">
                  <Link to="/" className="nav-link">Sign In</Link>
                </li>
                <li className="navbar-item">
                  <Link to="/signup" className="nav-link">Sign Up</Link>
                </li>
                <li className="navbar-item">
                  <Link to="/create" className="nav-link">Create Ticket</Link>
                </li>
                <li className="navbar-item">
                  <Link to="/track" className="nav-link">Track Ticket</Link>
                </li>
                <li className="navbar-item">
                  <Link to="/homepageUser" className="nav-link">Home</Link>
                </li>
                
              </ul>
            </div>
          </nav>

          <Route path="/" exact component={SignInForm} />
          <Route path="/signup" component={SignUpForm} />
          <Route path="/create" component={Ticktform} />
          <Route path="/track" component={TrackTicket} />
          <Route path="/homepageUser" component={homepageUser} />
        </div>
      </Router>

      </div>
    );
  }
}

export default App;


// old UI design
// class App extends Component {
//   render() {
//     return (
//     		<div className ="App">
//     			<div className= "App__Aside"></div>
//     			<div className=" App__Form">
//           			<BrowserRouter>
//           				<div>
//           					<Navigation />
//     						<Switch>
//     							<Route path = "/sign-in" component = {SignInForm} exact/>
//     							<Route path ="/Sign-Up" component ={SignUpForm} />
//     							<Route path = "/Ticktform" component ={Ticktform}/>
//     							<Route path ="/trackticket" component ={trackticket}/>
//     						</Switch>
//     					</div>
//     				</BrowserRouter>
//     			</div>
//     		</div>);
//   }
// }
