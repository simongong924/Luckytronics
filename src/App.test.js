
import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import renderer from 'react-test-renderer'
// import { exportAllDeclaration } from '@babel/types';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test("first snapshot test",() => {
  const component = renderer.create(
    <App />
  );
  let tree = component.toJSON();

  expect(tree).toMatchSnapshot();
})
