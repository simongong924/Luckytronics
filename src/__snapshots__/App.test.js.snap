// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`first snapshot test 1`] = `
<div
  id="parent"
>
  <div
    className="container"
  >
    <nav
      className="navbar navbar-expand-lg navbar-light bg-light"
    >
      <a
        className="navbar-brand"
        href="https://codingthesmartway.com"
        target="_blank"
      >
        <img
          alt="CodingTheSmartWay.com"
          height="30"
          src="logo.png"
          width="30"
        />
      </a>
      <a
        className="navbar-brand"
        href="/"
        onClick={[Function]}
      >
        Ticket System
      </a>
      <div
        className="collpase nav-collapse"
      >
        <ul
          className="navbar-nav mr-auto"
        >
          <li
            className="navbar-item"
          >
            <a
              className="nav-link"
              href="/"
              onClick={[Function]}
            >
              Sign In
            </a>
          </li>
          <li
            className="navbar-item"
          >
            <a
              className="nav-link"
              href="/signup"
              onClick={[Function]}
            >
              Sign Up
            </a>
          </li>
          <li
            className="navbar-item"
          >
            <a
              className="nav-link"
              href="/create"
              onClick={[Function]}
            >
              Create Ticket
            </a>
          </li>
          <li
            className="navbar-item"
          >
            <a
              className="nav-link"
              href="/track"
              onClick={[Function]}
            >
              Track Ticket
            </a>
          </li>
          <li
            className="navbar-item"
          >
            <a
              className="nav-link"
              href="/homepageUser"
              onClick={[Function]}
            >
              Home
            </a>
          </li>
        </ul>
      </div>
    </nav>
    <div
      className="FormCenter"
    >
      <h2>
         Sign In
      </h2>
      <form
        className="FormFields"
        onSubmit={[Function]}
      >
        <div
          className="FormField"
        >
          <label
            className="FormField__Label"
            htmlFor="email"
          >
            E-Mail Address
          </label>
          <input
            className="FormField__Input"
            id="email"
            name="email"
            onChange={[Function]}
            placeholder="Enter your email"
            type="email"
            value=""
          />
        </div>
        <div
          className="FormField"
        >
          <label
            className="FormField__Label"
            htmlFor="password"
          >
            Password
          </label>
          <input
            className="FormField__Input"
            id="password"
            name="password"
            onChange={[Function]}
            placeholder="Enter your password"
            type="password"
            value=""
          />
        </div>
        <div
          className="FormField"
        >
          <button
            className="FormField__Button mr-20"
          >
            Sign In
          </button>
           
        </div>
      </form>
      <div
        data-onloadcallbackname="onloadCallback"
        data-verifycallbackname="verifyCallback"
        id="g-recaptcha"
      />
    </div>
  </div>
</div>
`;
